### ISSUE TITLE
VimwikiGenerateTags and VimwikiDiaryGenerateLinks end up with empty wiki files

### ISSUE THREAD

**Issue by vrdel** [vimwiki/vimwiki#921]
Prior to submitting a new issue make sure to complete these steps:

- [x] Checkout the `dev` branch and confirm the issue is present there as well.
      The `dev` branch contains fixes that may not have been merged to `master` yet.
- [x] Post the syntax you are using (default/mediawiki/markdown) **and** your vimwiki settings from your `.vimrc`
- [x] Provide a detailed description of the problem including **steps to reproduce the issue**.
- [x] Include the output of `:VimwikiShowVersion`.

Hi!
After update to new stable version v2.5, I cannot generate diary index nor list of tags with references anymore. Both functions delete all current entries in wiki file so that it ends up completely empty.

I'm using default syntax and relevant entries from my `.vimrc` wrt vimwiki configuration:
```
" vimwiki stuff
let g:vimwiki_folding='expr'
let wiki = {}
let wiki.path = '~/my_notes/wiki'
let wiki.path_html = '~/Dropbox/wiki_html'
let wiki.auto_export = 0
let wiki.template_path = '~/.vim/myscripts/'
let wiki.template_default = 'template'
let wiki.template_ext = '.tpl'
let wiki.nested_syntaxes = {'py': 'python', 'rb': 'ruby', 'pl': 'perl', 'c': 'c',
                    \ 'html': 'html', 'js': 'javascript', 'css': 'css', 'xml': 'xml',
                    \ 'sh': 'sh', 'zsh': 'zsh',
                    \ 'sed': 'sed', 'awk': 'awk', 'lex': 'lex',
                    \ 'markdown': 'markdown', 'otl': 'votl', 'tex': 'tex',
                    \ 'man': 'man', 'help': 'help', 'vim': 'vim', 'conf': 'config'}
let g:vimwiki_list = [wiki]
let g:vimwiki_ext2syntax = {'.wiki': 'media'}
let g:vimwiki_hl_headers = 1
let g:vimwiki_list_ignore_newline = 1
let g:vimwiki_table_auto_fmt = 0

function! VimwikiWikiIncludeHandler(value)
	pyfile ~/.vim/myscripts/includehandle.py
endfunction

function! VimwikiLinkHandler(link)
	pyfile ~/.vim/myscripts/linkhandle.py
endfunction

let g:vimwiki_toc_header = 'Sadrzaj'
```
Erroneous behavior is easy to produce, at least here on my two workstations with identical vim setup. For diary index, I just point the editor to VimwikiDiaryIndex and fire up `:VimwikiDiaryGenerateLinks`. Instead of updated list of month with each days that have diary entries, I end up with empty diary index, deleting all entries that were previously there. Exactly the same behaviour I'm experiencing with `:VimWikiGenerateTags`. I point the editor to wiki file with all tags and references, fire up `:VimWikiGenerateTags` and instead of updated tags references, I end up with empty wiki file, deleting all tags and references that were previously there.

Vimwiki version that is misbehaving is (output of `:VimwikiShowVersion`): `Stable version: 2.5`
I'm running custom compiled vanilla vim: `version 8.2.834`

**Comment by tinmarino** [comment]
It is due to the fold method, a strange side effect : `set foldlevel=30` can save the queen or removing : `let g:vimwiki_folding='expr'`

minimal reproductible vimrc:

```
let g:vimwiki_folding='expr'
let wiki = {}
let wiki.path = '~/Test/WikiDefault'
let wiki.path_html = '~/Test/WikiDefault_html'
let g:vimwiki_list = [wiki]

packadd wiki
" This lines saves the queen
" set foldlevel=30
```

**Comment by tinmarino** [comment]
Fixed: af08e34 ? I updated the dev branch, can you check ?

It temporary "just" setlocal foldlevel=100 which has a lot of side effect especially because the update_listing_in_buffer loop is measuring if the fold is open or close.

Thanks @sdondley
